arangod/Agency/Node.cpp

```diff
@@ -201,14 +218,40 @@ Node& Node::operator=(Node const& rhs) {
   // Must not move rhs's _parent, _store
   removeTimeToLive();
   _nodeName = rhs._nodeName;
-  _children.clear();
-  for (auto const& p : rhs._children) {
-    auto copy = std::make_shared<Node>(*p.second);
-    copy->_parent = this;  // new child copy has us as _parent
-    _children.insert(std::make_pair(p.first, copy));
-  }
-  _value = rhs._value;
-  _vecBuf = rhs._vecBuf;
+  if (rhs._children) {
+    if (_children) {
+      _children->clear();
+    } else {
+      _children = std::make_unique<Children>();
+    }
+    for (auto const& p : *rhs._children) {
```

Comment:
here we could again reserve enough space for the children to come.

Reply:
Done.